### ISSUE TITLE
LSC Indoor IP Camera Firmware v7.6.32

### ISSUE THREAD

**Issue by OfficialDevvCat** [guino/Merkury1080P#37]
Hey there!

I have bought this LSC Indoor IP Camera on the 30th of August 2022 and tried this method (combined with the Merkury720P method) with no success.

[<img src="https://action.com/hostedassets/CMSArticleImages/72/96/3200019_8712879155065-111_01.png">](https://action.com/hostedassets/CMSArticleImages/72/96/3200019_8712879155065-111_01.png)

I have literally tried everything that was stated in the documentation. Also switching between SD cards. I also ready some other issues but nothing seems to help. I even tried the custom QR code that somebody in the issues stated but no lucky. I think they have patched out some things in this firmware version as this one is pretty high compared to all the other versions I saw wondering on Github.

The only ports that are open are:
- 80 (DoorBird video doorbell rtspd)
- 835
- 6668
- 8554 (DoorBird video doorbell rtspd)

Port 80 and 8554 showed "version" `DoorBird video doorbell rtspd` in nmap. I have no idea why it is also saying that on port 80 as that should be an HTTP server.

I also get no positive response from the HTTP requests I'm doing. I tried the `admin:admin` but also `admin:056565099`. They all returned `REFUSED`. I checked the SD card but no new folders or files have been created.

It's a cheap camera with a pretty decent lens on it and would love to see this work in my setup. I do NOT want to build one my own (for cheap) or buy an expensive set.

If you have any idea what I can do, let me know! :)

**Comment by guino** [comment]
@OfficialDevvCat if the Merkury1080, Merkury720 and BazzDoorBell process didn't work with different SD cards then it may have a different address or not be linux OS. Usually ppsFactoryTool.txt allows the HTTP responses to work, but like you said, it is possible they closed some things or changed the user/password.

Until someone with the right tools can open it up and read the firmware (or connect to UART) we won't know -- I have the tools but no device, so I have no way of helping right now.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat if the Merkury1080, Merkury720 and BazzDoorBell process didn't work with different SD cards then it may have a different address or not be linux OS. Usually ppsFactoryTool.txt allows the HTTP responses to work, but like you said, it is possible they closed some things or changed the user/password.
> 
> Until someone with the right tools can open it up and read the firmware (or connect to UART) we won't know -- I have the tools but no device, so I have no way of helping right now.

100% true! What do you exactly need to use the uart port? I have soldering skills, yet I have no idea what I need. Would a Raspberry Pi 4 work? Because I have that laying around. Let me know. Maybe you want to discuss this on discord? DevvCat#0880

**Comment by guino** [comment]
@OfficialDevvCat if you're willing to open your device and solder wires to it I can help trying to figure out if we can root it.

The first step would be to open your device and take some good pictures of the board so we can identify the UART pins.

The second step will be to solder some wires to the UART pins and connect GND, RX and TX to the GND, RX, TX of the pi board (GPIO15 and GPIO14 pins on the header). You may need to swap the RX/TX around as we won't know which is RX/TX by looking at the board.

Once you have it all connected you should see some messages on the pi terminal when you power on the device -- ideally you should be able to interrupt the boot of the device by pressing a key when the first messages show up and it will either give you a bootloader prompt or ask for a password. Whatever messages show up may help in figuring out if we can even do anything.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat if you're willing to open your device and solder wires to it I can help trying to figure out if we can root it.
> 
> The first step would be to open your device and take some good pictures of the board so we can identify the UART pins.
> 
> The second step will be to solder some wires to the UART pins and connect GND, RX and TX to the GND, RX, TX of the pi board (GPIO15 and GPIO14 pins on the header). You may need to swap the RX/TX around as we won't know which is RX/TX by looking at the board.
> 
> Once you have it all connected you should see some messages on the pi terminal when you power on the device -- ideally you should be able to interrupt the boot of the device by pressing a key when the first messages show up and it will either give you a bootloader prompt or ask for a password. Whatever messages show up may help in figuring out if we can even do anything. 

Okay! I'm 100% be willing to open up the device but currently I'm in the middle of a move so that have to wait until I've got all my stuff! I will let you know as soon as possible when I got everything and ready to the hacking! :)

**Comment by guino** [comment]
![Screenshot_2022-09-13_16-11-49](https://user-images.githubusercontent.com/4961810/190000509-a03536db-2e31-465e-b8da-713e369e3003.png)

Flash Chip is in marked in RED
UART pins are marked in BLUE

Either we need a copy of the firmware (using hardware programmer on the flash chip) OR we need someone to connect to the UART pins (TTL 3V) to capture an output log and/or see if there's any access to the bootloader.

**Comment by OfficialDevvCat** [comment]
> ![Screenshot_2022-09-13_16-11-49](https://user-images.githubusercontent.com/4961810/190000509-a03536db-2e31-465e-b8da-713e369e3003.png)
> 
> Flash Chip is in marked in RED UART pins are marked in BLUE
> 
> Either we need a copy of the firmware (using hardware programmer on the flash chip) OR we need someone to connect to the UART pins (TTL 3V) to capture an output log and/or see if there's any access to the bootloader.

I ordered a SPI flasher and it will arrive in a few hours. When it is here I can dump the firmware and upload it so you guys can take a look at it! Give me a few hours and I will get back to you guys.

To be specific, this is the one I ordered: https://www.amazon.nl/gp/product/B08TVNPTQK/

**Comment by guino** [comment]
@OfficialDevvCat I have that flash programmer and you need to be aware of 2 things:
1-It says it is compatible with 3.3V chips but it outputs 5V on some pins -- you should check it before you fry your chip/board. There's a 'mod' you can do on it to make it output 3.3V on all pins (I did the mod on mine and it works correctly)
2-You most likely will need to remove the chip from the board (OR at least disconnect PIN 6 in my experience) before you can read/write the flash. If you don't have a heat gun it may be easier to cut the pin 6 (with needle cut pliers) and solder it back afterward than trying to remove the whole chip (trying to disconnect pin 6 with a soldering iron will likely damage the board) -- learned the hard way.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat I have that flash programmer and you need to be aware of 2 things:
> 1-It says it is compatible with 3.3V chips but it outputs 5V on some pins -- you should check it before you fry your chip/board. There's a 'mod' you can do on it to make it output 3.3V on all pins (I did the mod on mine and it works correctly)
> 2-You most likely will need to remove the chip from the board (OR at least disconnect PIN 6 in my experience) before you can read/write the flash. If you don't have a heat gun it may be easier to cut the pin 6 (with needle cut pliers) and solder it back afterward than trying to remove the whole chip (trying to disconnect pin 6 with a soldering iron will likely damage the board) -- learned the hard way.
> 
> **IF you're going to do any cut/solder work: I recommend practicing on any old/broken board laying around first**

Okay so wait.
The camera needs to be turned on right? And then the clip needs to be attached before i turn it on. (If it outputs the same voltage) and then i need them both turned on and read the data? Or does rhe camera needs to be off and then attach the clip and read the data? Cause if the clip outputs 5v. I can make sure the clip does not output power and then connect the clip and turn on the camera to extract the data. Or is that not going to work? I do not have a heatgun or soldering station on hand.

**Comment by guino** [comment]
@OfficialDevvCat a flash programmer will read (and later write - if desired) the built in firmware on the device -- to be clear: it won't do anything with UART. For reading/writing the flash you don't turn on the device at all, you just plug the flash programmer on the chip and read (or write) its contents (like a USB drive). The issue is that (from experience) connecting the flash programmer to the chip without removing it from the board doesn't work (fails to read/write). As long as you verify the output is 3.3V (on all pins like VCC, RX, TX) then it should be safe to try and read the flash while connected to the board (but from experience it is likely going to fail, but who knows board design changes). If you plug the programmer to the chip while on board and you output 5V to any pin you may fry the device (fair warning).

The only type of connection we do with the board/device powered ON is when using the UART/TTL adapter where we connect it then power on the device to capture the boot output log. UART connections require a USB/TTL UART/SERIAL adapter (3.3V), which is a different thing than the flash programmer.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat a flash programmer will read (and later write - if desired) the built in firmware on the device -- to be clear: it won't do anything with UART. For reading/writing the flash you don't turn on the device at all, you just plug the flash programmer on the chip and read (or write) its contents (like a USB drive). The issue is that (from experience) connecting the flash programmer to the chip without removing it from the board doesn't work (fails to read/write). As long as you verify the output is 3.3V (on all pins like VCC, RX, TX) then it should be safe to try and read the flash while connected to the board (but from experience it is likely going to fail, but who knows board design changes). If you plug the programmer to the chip while on board and you output 5V to any pin you may fry the device (fair warning).
> 
> The only type of connection we do with the board/device powered ON is when using the UART/TTL adapter where we connect it then power on the device to capture the boot output log. UART connections require a USB/TTL UART/SERIAL adapter (3.3V), which is a different thing than the flash programmer.

Okay that is clear. Could you take a look at the picture and tell me what the best solution could be? It has a jumper for maybe possible ttls fu functinality? 

https://imgur.com/a/I0ZPCF0

**Comment by guino** [comment]
@OfficialDevvCat the chip should connect on the ‘25’ section, but like I said: this programmer has a 3.3V/5V jumper but when you set it to 3.3V it still outputs 5V on some pins ( RX/TX pins I think ). If you connect it without the mod to fix the voltage you may damage the board/chip (you have been warned).

**Comment by OfficialDevvCat** [comment]
Okay, I will find online if there a different way to do that. What about the TTL functionality? Does that also output 5v? Or is it just a reading pin?

**Comment by guino** [comment]
@OfficialDevvCat that flash programmer only has TTL functionality - in 3.3V or 5V selected by jumper switch, but this is only for flash chip read/write. The UART pins require a UART TTL (3.3V) adapter which is mostly available as a USB adapter. There’s no way (that I know) to use a flash programmer on the UART TTL pins, and there’s no way (that I know) to use the he UART TTL adapter on the flash chip.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat that flash programmer only has TTL functionality - in 3.3V or 5V selected by jumper switch, but this is only for flash chip read/write. The UART pins require a UART TTL (3.3V) adapter which is mostly available as a USB adapter. There’s no way (that I know) to use a flash programmer on the UART TTL pins, and there’s no way (that I know) to use the he UART TTL adapter on the flash chip.

I just check the datasheet of a old 25xx chip from a dead GPU and it was a 3.6 max volt chip. And it i got to read the chip without issues and even got to write to it. I also looked up and this is an improved revision of the board with the 3.3v fix. Should I now extract the data from the camera?

**Comment by guino** [comment]
@OfficialDevvCat sounds good if you say it’s a fixed version you can try - again, from my experience it may not work while the chip is soldered on the board. I would avoid keeping the flash programmer hooked up for a long period just in case the voltage is wrong, so hook it up, try to read, remove it if fails, wait a bit hook up, try reading again, etc

**Comment by OfficialDevvCat** [comment]
I just dumped the chip, You can download the dumped bin file from here https://www.mediafire.com/file/31ms1k4kgqxxlh6/Smart_Indoor_IP_Camera.bin/file
Let me know what I can do or what you're planning to do.

**Comment by guino** [comment]
@OfficialDevvCat well, assuming your camera still works normally, I would try this first:

Follow the steps from:
https://github.com/guino/Merkury1080P#conclusion USING THE ATTACHED 3 files: [7632.zip](https://github.com/guino/Merkury1080P/files/9587283/7632.zip) -- that is, instead of what's posted on the link (I changed the address changed to A0008000 on env and ppsMmcTool.txt files). Assuming the internals didn't change a lot this may allow you to root device.

Binwalk didn't give me a lot of information to work with, so let's hope this works.

**Comment by OfficialDevvCat** [comment]
@guino I already tried that the first time. But that didn't work. Wasn't that the main reason why I dumped the chip?

**Comment by OfficialDevvCat** [comment]
@guino Just one more question. Do I have to flash the chip afterwards to make it work?? Cause the issue is that the SPI Flasher does not work anymore for some reason?! The Red power light is on and so is the Yellow RUN led. And it is not found by windows anymore?! I don't hear a USB Connected sound. Very very strange. So I'm going to return it tomorrow and ask for a new one which will take a few days... Unless you know what I can do about it?

**Comment by guino** [comment]
@OfficialDevvCat If you tried A0008000 and it didn't work I'll have to try and dig out more from this firmware file. 

Is the camera still working ? (boots up, etc) ? if not you may have damaged it somehow. It may just need a power cycle of your machine to reset the USB bus (if it used too much power).

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat If you tried A0008000 and it didn't work I'll have to try and dig out more from this firmware file.
> 
> Is the camera still working ? (boots up, etc) ? if not you may have damaged it somehow. It may just need a power cycle of your machine to reset the USB bus (if it used too much power).

Camera works perfectly fine! I can try rebooting my PC but I don't think it's really going to work as it was plugged in a powered hub. Tried a different pc and used my powerbank. Yellow light stays solid and not connected.

**Comment by OfficialDevvCat** [comment]
But again, Do I need to flash stuff again onto the chip? Cause I can't read anywhere what to do after the edit. I assume it has to

**Comment by guino** [comment]
@OfficialDevvCat the 'Read' process doesn't change anything in the chip, so to work 'normally' you won't have to flash anything back. If I can unpack the firmware we may be able to find something to change to root the device (I haven't been able to do it yet) -- in that case you would need to be able to write the changes with the programmer (meaning it would need to work again).

**Comment by OfficialDevvCat** [comment]
I tried to find some variables from the link you sent me but it can't even find the "Loadable segment". Maybe you could take a look at it if you've got the time for it? Would appreciate it!

**Comment by OfficialDevvCat** [comment]
@guino Let me know if I can do anything as I'm a programmer and know my way around some of this stuff! Would be very cool to get this thing streaming a signal outside the app. :) Have a great weekend in advance and lets hope for some good results on this thing

**Comment by guino** [comment]
@OfficialDevvCat I will try to see if I can get anything out of it - may need to try a different tool.

**Comment by guino** [comment]
@OfficialDevvCat I downloaded a different tool and then I noticed there's an issue with your flash file -- it's only 2Mb when it should be at least 8Mb (some devices have 16Mb) -- this is likely the reason why I could not extract anything out of it. This may have been something like selecting the wrong size of chip when you did the 'read' or perhaps an issue identifying the chip size (or even just an upload issue), Do you happen to have the /proc/cmdline for this device ? thay may help me extract the bootloader from the section you provided (so I can try to double check the load address).

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat I downloaded a different tool and then I noticed there's an issue with your flash file -- it's only 2Mb when it should be at least 8Mb (some devices have 16Mb) -- this is likely the reason why I could not extract anything out of it. This may have been something like selecting the wrong size of chip when you did the 'read' or perhaps an issue identifying the chip size (or even just an upload issue), Do you happen to have the /proc/cmdline for this device ? thay may help me extract the bootloader from the section you provided (so I can try to double check the load address).

I did not have any success with that sadly. The http server is sadly disabled to get any kind of useful info from it.

U have to wait until sunday until I got my new reader to try it out once again.

**Comment by OfficialDevvCat** [comment]
@guino I just received my new SPI flasher. I will dump the whole chip (8MB) in an hour or so. I will lookup the chip model number and see what settings I need.

**Comment by OfficialDevvCat** [comment]
@guino I just noticed the TX/RX rail is on 4.5V. I will have to wait until Wednesday till I got my soldering station.

**Comment by OfficialDevvCat** [comment]
@guino I just made a new dump! Desoldered the chip as I was having issues. And dumped the whole 8MByte/64MBit https://www.mediafire.com/file/toofgq5d16pgdf5/ipcam_new.bin/file

**Comment by guino** [comment]
@OfficialDevvCat At a quick first look this device seems very different than what I have seen before -- I'll have to dig in with more time to see if there's anything we can do -- since you have a programmer and the chip outside, I likely can modify the firmware so you can flash it back and have it 'rooted' but I'm not sure what all is available on it yet. I assume you'd like to enable RTSP if possible ? I also assume you have tried the normal rtsp urls like rtsp://IP:8554 and had no success ?

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat At a quick first look this device seems very different than what I have seen before -- I'll have to dig in with more time to see if there's anything we can do -- since you have a programmer and the chip outside, I likely can modify the firmware so you can flash it back and have it 'rooted' but I'm not sure what all is available on it yet. I assume you'd like to enable RTSP if possible ? I also assume you have tried the normal rtsp urls like rtsp://IP:8554 and had no success ?

I would like to have rtsp only. I think thats the most simple and will use external software to manage it all. I did try rtsp urls and that didn't work eighter. If you want we can go into a discord call (if you have the time) so we both can take a look at it and talk about it if you want?

**Comment by guino** [comment]
@OfficialDevvCat Can you try using https://sourceforge.net/projects/onvifdm/ with user **admin** and password **12345678** ? may be worth trying the RTSP urls with **admin:12345678** as well? I would also try **admin** with password **123456** (on onvif and rtsp) as I have seen references to these in the code. Additionally if there's any username/password you configured in the app (for anything) I'd try that as well (and with **admin** and whatever password).

You may also want to check if this works: https://ipc-us.ismartlife.me/login (scan it with your tuya app on your 'Me' section) -- I remember reading about streaming from that interface.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat Can you try using https://sourceforge.net/projects/onvifdm/ with user **admin** and password **12345678** ? may be worth trying the RTSP urls with **admin:12345678** as well? I would also try **admin** with password **123456** (on onvif and rtsp) as I have seen references to these in the code. Additionally if there's any username/password you configured in the app (for anything) I'd try that as well (and with **admin** and whatever password).
> 
> You may also want to check if this works: https://ipc-us.ismartlife.me/login (scan it with your tuya app on your 'Me' section) -- I remember reading about streaming from that interface.

So you want me to solder the chip back on and try that? 

Btw, i cannot add it to the tuya app. I tried but it didn't work

**Comment by guino** [comment]
@OfficialDevvCat If the chip is out, lets hold out on that request. I'm still trying to find something we can try to change on the firmware.

On the https://ipc-us.ismartlife.me/login -- when I log in with the tuya app it shows my cameras in it, but it shows 'No WebRTC support' for both of them -- so it logs in but my devices don't support it. What did you get when you tried to login ? if you're in europe you may need to use this one instead: https://ipc-eu.ismartlife.me/ -- any information on the exact issue you get would be helpful -- for now I'm interested in knowing it it shows 'No WebRTC supported' when you hover your camera name.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat If the chip is out, lets hold out on that request. I'm still trying to find something we can try to change on the firmware.
> 
> On the https://ipc-us.ismartlife.me/login -- when I log in with the tuya app it shows my cameras in it, but it shows 'No WebRTC support' for both of them -- so it logs in but my devices don't support it. What did you get when you tried to login ? if you're in europe you may need to use this one instead: https://ipc-eu.ismartlife.me/ -- any information on the exact issue you get would be helpful -- for now I'm interested in knowing it it shows 'No WebRTC supported' when you hover your camera name.

The thing is that i have my camera not connected to the tuya app. I used the LSC app and i removed it on there as well... I can still try to login but not a lot would happen

**Comment by guino** [comment]
@OfficialDevvCat most tuya compatible apps are basically just a rebrand of the tuya app (have similar look and feel, features, etc). On the tuya app I click on the "Me" section at the bottom as seen here:
![image](https://play-lh.googleusercontent.com/UkGCWYXfyYw3sYKnDvKe3sRW78GhD0sQ40DPXrvw6fJueBTJr_o9iLkbXVJHCWDVRDk=w5120-h2880) 

then click on a little bar code scanning icon at the top like on this image:
![image](https://user-images.githubusercontent.com/4961810/192040318-77291d9d-785f-44f8-bf0b-cef8da165a05.png) 

 that's where I scan the QR for login on https://ipc-eu.ismartlife.me/

If you don't have that on the lsc app we may need to wait till you solder the chip back to remove the device from the lsc app and try it on the tuya app (by enrolling it on it).

**Comment by OfficialDevvCat** [comment]
@guino we're in luck! The camera is still in the LSC app and I can scan the qr code. I will do that in an hour or so as >mI'm not home yet

**Comment by OfficialDevvCat** [comment]
I kust did it quickly on my phone and this is what I see https://imgur.com/a/xNmpztr

**Comment by guino** [comment]
@OfficialDevvCat If you hover on that text ‘LDC Smart Indoor Camera’ and it doesn’t display a popup like ‘WebRTC not supported’ then most likely you can use that page to view the camera feed when it’s online. I believe some people got a way to extract that feed and re-stream it elsewhere (my devices are not compatible so I don’t know much about it).

**Comment by OfficialDevvCat** [comment]
@guino i just tried the site on my pc and it said "Cross region login not supported" ...

**Comment by OfficialDevvCat** [comment]
@guino on my phone it does not seem it is showing a popup. I don't know for sure though but I want to look at it on the pc. I kinda want to solder the chip back on tomorrow and see if that page works and to see if I can connect it to tuya. But I don't want to use cloud based stuff... I want to use it localy if that is possible.

**Comment by EpicLPer** [comment]
I just got a LSC Outdoor one and haven't tried any of the methods yet, but I doubt it'll work. Just had to update the camera to even use it, but I'd also be willing to try and hack it since I have soldering skills and so on in case you need dumps or anything :)

**Comment by OfficialDevvCat** [comment]
> I just got a LSC Outdoor one and haven't tried any of the methods yet, but I doubt it'll work. Just had to update the camera to even use it, but I'd also be willing to try and hack it since I have soldering skills and so on in case you need dumps or anything :)
> 
> EDIT: Yeah, just tried and it didn't work. My firmware version is 2.10.36, but I'll create a separate issue for this model.

@EpicLPer 
I think you should wait for us to complete this one. It's most likely they patched all the LSC cameras the same way.

**Comment by guino** [comment]
@OfficialDevvCat  I will rebuild a firmware for you to try with the following:
1-root password changed
2-telnet started by default
3-patched ppsapp (to hopefully enable RTSP though I am not sure it will work as the code is all different)

I just have to find a bit of time to work on it.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat  I will rebuild a firmware for you to try with the following:
> 1-root password changed
> 2-telnet started by default
> 3-patched ppsapp (to hopefully enable RTSP though I am not sure it will work as the code is all different)
> 4-a command to call custom.sh from the sd card
> 
> I just have to find a bit of time to work on it.

Awesome! I would really appreciate it! If you got something. Let me know so we can test it out! :)

**Comment by guino** [comment]
[ipcam_hack.zip](https://github.com/guino/Merkury1080P/files/9669153/ipcam_hack.zip)

The above firmware has a new rootfs with these changes:
1-added a user/password: hack/telnet (root password is unchanged)
2-telnet should start by default on port 24 (no user password required)
3-patched ppsapp so it force-runs the RTSP start code (unsure if it will make a difference)
4-a command to call custom.sh from the SD card every 10 seconds (like other hacks I made)

I found that there's possibly a way to enable telnet on these devices by creating a file called `product.cof` -- possibly with a content like this:
```
[DEFAULT_SETTING]
telnet=1
```
It could be confirmed by checking if port 23 opens with the file above on the SD card's root during boot.

Try it out when you have a chance and let me know what you find -- if for some reason your device doesn't boot I'd advise trying to flash your original firmware to see if it boots (I understand this requires de-solder/soldering but there's not much else we can do to check if the issue is the hardware or the firmware I made).

**Comment by OfficialDevvCat** [comment]
Thank you very very much! I will be flashing it right away and putting back the chip on the system! I will notify you by editing this message!

**Comment by guino** [comment]
@OfficialDevvCat it sounds like you could be having an issue with cable, programmer or chip. We have your original flash, writing should have been as easy as reading it. I am unfamiliar with the tools you’re using, but if it is not able to verify the data it is likely not writing it correctly - you may want to try selecting a different 25/64 chip from the list if available. You may also want to check the voltage on the pins - if it has 5v on any pin it could have damaged the chip. These 25/64 spi flash chips are very common and mostly compatible so if that chip is bad you could potentially try to get a new one to replace it, they are also common on some motherboards so you could find it on scrap boards. I would not bother soldering it back unless you are able to write and verify it.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat it sounds like you could be having an issue with cable, programmer or chip. We have your original flash, writing should have been as easy as reading it. I am unfamiliar with the tools you’re using, but if it is not able to verify the data it is likely not writing it correctly - you may want to try selecting a different 25/64 chip from the list if available. You may also want to check the voltage on the pins - if it has 5v on any pin it could have damaged the chip. These 25/64 spi flash chips are very common and mostly compatible so if that chip is bad you could potentially try to get a new one to replace it, they are also common on some motherboards so you could find it on scrap boards. I would not bother soldering it back unless you are able to write and verify it.

I got it working. FINALLY! I erased the chip and reprogramed it. Gonna desolder it now and putting back on the board. I will let you know how it goes! Do i have to make that file on the sd card in order to mske telnet work?

**Comment by guino** [comment]
@OfficialDevvCat no, the product.cof file is just something to try later - Telnet should be enabled by default on the firmware I posted (on port 24). If you have time to try later that product.cod file may enable the normal Telnet on port 23 (this would help people without a hardware programmer to root the device).

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat no, the product.cof file is just something to try later - Telnet should be enabled by default on the firmware I posted (on port 24). If you have time to try later that product.cod file may enable the normal Telnet on port 23 (this would help people without a hardware programmer to root the device).

I flashed the hacked file and soldered the chip. First it didn't work, was caused my a loose pin. Soldered it better on the board it it worked and has been set up in the LSC app. So it is not dead YET! haha. The only thing is that the screen is a bit purple ish for some reason?! I'm gonna test the telnet thing now, if that works we're in! I will be buying a new factory default camera today or tomorrow to test if the sd card thing works!

**Comment by guino** [comment]
@OfficialDevvCat I would not buy a new camera just to try the telnet thing -- for one: you can test if the telnet thing would work with your existing camera, second: I don't know the default root password, it may be hard to find it -- someone with a powerful GPU and john the ripper may be able to find it (I only have a laptop without powerful GPU in it). The default passwd file is this:
```
root:IsHLvr/zaeG2U:0:0:root:/:/bin/sh
```

**Comment by EpicLPer** [comment]
Can offer mine if you guys need one and tell me how to :) Have a 3080 Ti.

**Comment by guino** [comment]
@OfficialDevvCat nothing I changed software wise would be capable of changing color/tone on the picture -- this would have to be something hardware side (lens/sensor/etc). You obviously can load the original firmware back in the chip and verify that won't change the picture in any way, but it would be kind of wasted time.

I believe I may have made a mistake on the command to start telnet automatically -- so that's why that didn't start. The product.cof file I wasn't sure if it was going to work anyway.

I wanted you to try creating a custom.sh file in the root of the SD card with these contents:
```
if [ ! -e /tmp/customrun ]; then
 echo custom > /tmp/customrun
 /sbin/telnetd -l /bin/sh -p 25
fi
```
Then see if telnet works on port 25: `telnet IP 25`

One last thing: if you connecte to the UART the script that starts the camera reads from the standard input and if you keep pressing **n** it may not start the camera software (originally could throw you into the prompt) -- you could try to press CTRL+C to see if you get a command prompt anyway on the UART.

**Comment by OfficialDevvCat** [comment]
@guino Okay clear I will TRY that but I'm not sure if it will work. Lets hope for the best. I also saw this in UART
```
/usr/init/app_init.sh: line 41: /bin/telnetd: not found
```
I will update with anything I got

**Comment by guino** [comment]
@OfficialDevvCat that not found error is exactly why telnet didn't start -- it should have bin /sbin/telnetd instead of /bin/telnetd

**Comment by OfficialDevvCat** [comment]
@guino Pressing `n` or `CTRL+C` did not cancel anything sadly. I did saw something like this aswell
```
(We use port 80 for optional RTSP-over-HTTP tunneling.)
```
I will include a file of all the output the UART is outputing. Let me know if you found something interesting.
https://pastebin.com/u01yqLer

**Comment by guino** [comment]
@OfficialDevvCat you can add whatever you want in the SD card (assuming the application doesn't reformat it).

The file I made a mistake on is `/usr/init/app_init.sh` -- but this is in the flash memory so unless you re-flash it there will not be any way to modify it -- you can safely ignore that error.

Did you try the custom.sh file at all ? you don't need to do it if the telnet is already working, I'm just curious to see if it would work.

**Comment by guino** [comment]
@OfficialDevvCat Additionally, is your WHOLE camera feed purple ? or just part of it -- if it is not just part of the picture, then it could be the IR filter usially in these devices that is stuck in the wrong position ? maybe you forgot to connect the wire or connected it inverted ? just thinking or possible issues.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat you can add whatever you want in the SD card (assuming the application doesn't reformat it).
> 
> The file I made a mistake on is `/usr/init/app_init.sh` -- but this is in the flash memory so unless you re-flash it there will not be any way to modify it -- you can safely ignore that error.
> 
> Did you try the custom.sh file at all ? you don't need to do it if the telnet is already working, I'm just curious to see if it would work.

I did try the custom.sh file but did not do anything sadly. The product.cof did work though! And u said that that file was used in the factory flash.

**Comment by OfficialDevvCat** [comment]
@guino could you possible explain me how you changed the firmware? As I bought another device but its a  it different from the once in the thread picture.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat I'll send you the commands I used (I need to double check them to make sure nothing is wrong). The RTSP change in dgiot (which I'm not sure worked) is the one that is a bit harder to explain.

Would be very appreciated! As I got the LSC smart Rotateable Camera now!

**Comment by OfficialDevvCat** [comment]
![](https://www.action.com/_next/image/?url=https%3A%2F%2Faction.com%2Fhostedassets%2FCMSArticleImages%2F74%2F27%2F3007325_8712879151210-111_03_20220823124939.png&w=640&q=75)

This is the camera, Scanned all the ports and only port 6668 is open, same like the other one.

**Comment by guino** [comment]
@OfficialDevvCat did you try product.cof file by itself ? it may enable telnet -- problem is finding the password (you can try from known lists like this one: https://gist.github.com/gabonator/74cdd6ab4f733ff047356198c781f27d )

**Comment by OfficialDevvCat** [comment]
@guino I did not try that YET. I do want to mention something that has not been mentioned anywhere else. Check out this link https://developer.tuya.com/en/docs/iot/webrtc?id=Kacsd4x2hl0se I did everything what it said and i got a RTSPS stream working with sound! The only issue I'm having now is converting it into a RTSP stream. If that is somehow possible then I'm 100% set and I can make a repo for this stuff!

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat I believe that cloud approach only works on devices that have WebRTC support (i.e. mine don't).
> 
> There's no telling about the password -- different firmware versions, brand, models may have same or different passwords. My 2 devices (Merkury 720p and Bazz Doorbell) both had the same root password hash but I never found a working password.

Well im gonna try hashing the password. Could you possibly try to explain how u mounted the spi dump or maybe how u did it?

**Comment by guino** [comment]
@OfficialDevvCat the /etc/passwd file for the firmware you sent me before is:
`root:IsHLvr/zaeG2U:0:0:root:/:/bin/sh`

To extract the contents of the firmware I just use `binwalk -e -M flash.bin`

I know I still have to provide you with the commands I used to rebuild the firmware, it's not a lot (like 3-4 commands), I just have to re-do them to make sure they're correct (I usually don't memorize or save that stuff).

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat the /etc/passwd file for the firmware you sent me before is:
> `root:IsHLvr/zaeG2U:0:0:root:/:/bin/sh`
> 
> To extract the contents of the firmware I just use `binwalk -e -M flash.bin`
> 
> I know I still have to provide you with the commands I used to rebuild the firmware, it's not a lot (like 3-4 commands), I just have to re-do them to make sure they're correct (I usually don't memorize or save that stuff).

Allrighty! I will try that on the bin file that i still have. And I shall wait on those commands. I will also try to find stuff online in the meantime.

Also, there is an update for that camera available but it does not update for some reason. So it is probably too out of date for the `product.cof` file as that didn't work sadly. Not sure if it 100% didn't work or if I did something wrong. 

I'm also in the middle of a move (again). It will happen in about a week or 2.

**Comment by guino** [comment]
@OfficialDevvCat if you have a powerful machine this one should be able to try and find the password: https://github.com/openwall/john -- my laptop is not nearly enough to do it.

**Comment by guino** [comment]
@OfficialDevvCat so here is how I built your image:

1-Unpacked the firmware file with `binwalk -e -M flash.bin`
2-go into the extracted `squashfs-root` directory and make any changes you need -- you should make only minor changes. If you try adding files or adding too many commands on existing scripts you may end up not having space for it in the firmware, so keep changes to a minimum. On your firmware I changed etc/passwd (1 line added), usr/init/app_init.sh (10 lines added) and usr/bin/dgiot (patched a few bytes to try and enable RTSP -- no change in size).
3-Rebuild the squashfs-root data with the command below (from the directory where you have squashfs-root):
`mksquashfs squashfs-root/ new.squashfs -comp xz` -- your newly created new.squashfs file should be the similar in size to the original one (260000.squashfs extracted by binwalk) -- going a bit over is usually ok (on your firmware it seems like they had about 800Kb of space empty but the more you add the more changes you have of something going wrong).
4-create a new flash file from your original one: `cp flash.bin newflash.bin`
5-copy the newly built squashfs into your newflash.bin: 
`dd if=new.squashfs of=newflash.bin bs=1 conv=notrunc oseek=2490368` (the 2490368 is the same as 260000 hex value -- which is the original position of the squashfs displayed in binwalk when extracting on step 1)

The newflash.bin should be ready to flash at that time.

**Comment by guino** [comment]
@OfficialDevvCat 

I was in Europe this past week and bought one of each of these cameras:
![cam1](https://camo.githubusercontent.com/31a795202fd64f16ef4a880a7d721f8527bb8e645da4ef442168fe2a0f2fe03b/68747470733a2f2f616374696f6e2e636f6d2f686f737465646173736574732f434d5341727469636c65496d616765732f37322f39362f333230303031395f383731323837393135353036352d3131315f30312e706e67)

![cam2](https://camo.githubusercontent.com/a26262eba0bc5b682dad1c2be2cf5f191dc3e96e2a138e6928e0530bb5bac07d/68747470733a2f2f7777772e616374696f6e2e636f6d2f5f6e6578742f696d6167652f3f75726c3d6874747073253341253246253246616374696f6e2e636f6d253246686f73746564617373657473253246434d5341727469636c65496d6167657325324637342532463237253246333030373332355f383731323837393135313231302d3131315f30335f32303232303832333132343933392e706e6726773d36343026713d3735)

![cam3](https://user-images.githubusercontent.com/8442196/192149609-a1c3d70a-bea6-484f-955d-09eca376161f.png)

I'll do some more research on them as soon as I have some time.

**Comment by OfficialDevvCat** [comment]
@guino awesome! Thats cool! Where did you find them? I did not try to disassemble the rotsting camera yet because I'm still in the middle of a move. I will be moving permanently this Wednesday. After that I can play around with the rotating camera.

**Comment by guino** [comment]
@OfficialDevvCat I found them at an 'Action' store -- had to go look at 2 stores to find them. I honestly haven't even powered them on yet but I will likely add them to my TUYA (not LSC) app -- that usually makes a difference as to what firmware updates are available.

**Comment by EpicLPer** [comment]
> @OfficialDevvCat I found them at an 'Action' store -- had to go look at 2 stores to find them. I honestly haven't even powered them on yet but I will likely add them to my TUYA (not LSC) app -- that usually makes a difference as to what firmware updates are available.

Take a dump of it first tho :) Would do one for mine but I know those cams usually don't fully clear passwords or tokens properly.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat I found them at an 'Action' store -- had to go look at 2 stores to find them. I honestly haven't even powered them on yet but I will likely add them to my TUYA (not LSC) app -- that usually makes a difference as to what firmware updates are available.

Only my rotateable cam worked trough tuya instead of LSC app. Idk why. Both app ha e the same update and it didnt update. Was it in the netherlands possibly? I live there btw and also got then from the "action" store

**Comment by guino** [comment]
@OfficialDevvCat yes, Netherlands. I just added the outdoor camera to the Tuya app and it shows 2.10.22 version in the app --I have auto-upgrade turned off so it never updates without my permission.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat yes, Netherlands. I just added the outdoor camera to the Tuya app and it shows 2.10.22 version in the app --I have auto-upgrade turned off so it never updates without my permission.

Really! Interesting. What camera did you pick from the list?  Cause mine just does not even make the bro sound.

**Comment by guino** [comment]
@OfficialDevvCat I added it like this: 
-clicked + sign on top-left
-selected add device
-selected camera & lock (left side)
-selected 'Smart Camera (Wi-fi)
-followed the prompt to scan QR code while on 2.4Ghz network

That was all!

I also just verified the WebRTC works on this camera (it doesn't on my other cameras) -- it lets me select them and says 'this camera is unavailable on the web, please use phone app'.

**Comment by OfficialDevvCat** [comment]
@guino hmm okay. I will try again when I moved just to make sure. Do you have the tools to dump the chips? I assume so. Gonna be an interesting development. Also the rtsp stream is without audio sadly. And where in NL were you if I may ask? Did you enjoy the vacation?

**Comment by OfficialDevvCat** [comment]
@guino Okay I finally moved in and my whole network setup has been installed! I'm ready to do modifications and such! I really want to have audio trough the RTSP stream of the first camera we hacked.

**Comment by guino** [comment]
@OfficialDevvCat Sorry I haven't had a chance to do much more than plug my cameras in and verify they work -- I want to try and make the hostapd thing work directly (root) then I can focus on the application itself.

**Comment by OfficialDevvCat** [comment]
@guino Okay, what you mean with the `hostapd`? Is there anything I can help you with?

**Comment by guino** [comment]
@OfficialDevvCat sorry, hostapd is a file we're using to root the outdoor camera (getting confused here with all the cameras). I'll have to take a look at the code for the indoor camera to see about the RTSP audio.

**Comment by OfficialDevvCat** [comment]
@guino I just removed the chip from the rotating camera and dumped the chip multiple times. But it is missing data every single time (lots of FF bytes trough the data). `binwalk` also is not able to extract any of the data. I'm kinda stuck at this point.

**Comment by guino** [comment]
@OfficialDevvCat FF ususally is written to the flash to 'fill in' empty areas, but doesn't necessarily mean something is wrong. If you read the data 2 times and the md5sum matches chances are it is the correct data (assuming nothing is wrong with the chip/hardware). Feel free to post or email me the dump file and I can take a quick look. I have been dealing with other stuff at home so I haven't had much time to mess with cameras yet.

**Comment by OfficialDevvCat** [comment]
Okay, so the issue was that I was using a too long of a cable between the reader and the chip. Fixed it by putting it on a board. I dumped the chip and here it is: https://easyupload.io/p2r6ew (Yes it took me 9 dumps in order to get it correct haha). I'm not sure how and all this works cause this is a whole different structure compared to the last camera (Most likely cause it is not updated). But I did see something in `/squashfs-root-0/local/factory_cfg.ini`. Let me know if you find some interesting stuff!

Btw, i ripped 2 pins off the board while removing it.... I did fix it though but I'm not sure if it works. I think it will though.

**Comment by guino** [comment]
@OfficialDevvCat you can use just 'seek=1835008' instead of oseek= (basically the same parameter renamed). At a quick look the hostapd hack for the outdoor camera may work on this one too. The update process on these cameras seems fairly simple too, you basically create an update.tar file in the SD card with the right files and it will just overwrite the flash section for you, doesn't seem to have any special headers/signature, certificate checks or anything -- I would rather not use that because I like keeping the flash untouched if possible.

**Comment by OfficialDevvCat** [comment]
@guino allright! I will try that! I also moved eveything from `squashfs-root-0` to `squasfs-root` and then did the command to creat a new squasfs file. I hope that is fine? Cause it created 2 squasfs folders. Should I then create the new file with the changes I made and reflash it and solder it back on? Or you want me to wait for you to tell me to do something else?

**Comment by guino** [comment]
@OfficialDevvCat The name of the directory should not matter when building the squash-fs file -- but renaming should be fine. Sorry I'm at work and didn't see your message till now.

I use bs=1 because I'm lazy and don't want to calculate the values based on different block sizes. A higher bs value just makes the dd command 'faster', it doesn't affect the end-result (firmware file). You can send me your modified dump file if you like me to verify it before you flash it, but since you have the original you can always restore it (with some soldering work).

**Comment by OfficialDevvCat** [comment]
I modified it and flashed it. But 2 pins came off like i told u. Im trying to resolder the connections. If that does not work I will get a new one

**Comment by guino** [comment]
@OfficialDevvCat a lot of times you can 'trace' the connection to a nearby resistor or component around it and solder a new wire to it -- that's what I had to do with one of my devices, but I learned my lesson and will only remove chips with a heat gun now.

**Comment by OfficialDevvCat** [comment]
@guino Well tbh, idk what the issue is. It does not work. It could be hardware related (due to the 2 missing pins I tried to solder back on). or software related as I did flash a new firmware on the chip. I will be getting a new camera tomorrow (the first model we hacked as that one is better mountable and I do not need a rotatable camera anyway, it was fun to play with the rotatable one).

**Comment by OfficialDevvCat** [comment]
@guino i have a question. Is it possible to flash the hcked firmware on a new camera of the same model? I assume itis no problem right?

**Comment by guino** [comment]
@OfficialDevvCat Yes-you can flash the same firmware on multiple cameras IF they are the same exact hardware BUT they will all have the same factory identification (even if the firmware was dumped before phone app registration), which means they can’t be online at the same time or the tuya cloud will likely refuse connection. So if you intend to use all cameras online I would not do it, but if one camera died and you want to use the dumped flash on another one, yes it should be fine and the camera will basically be an exact ‘copy’ of the other one.

**Comment by guino** [comment]
@OfficialDevvCat I believe there are other backdoors, so I would hold off on the hardware route until I can have a look at it. Sorry it’s been a hectic couple of weeks here. 

There’s no easy way to explain what I changed in the main application, but if they’re the same we should be able to just update the main application with the one I modified. Ideally I want to make something that leaves the firmware as original as possible and runs our scripts/applications from the SD card like on other devices.

**Comment by OfficialDevvCat** [comment]
@guino ofcorse. We do not want to touch the hardware if it is not needed. I will unpack one camera. Connect it trough the app and NOT update it and check for ports. That is all I can do atm without updating.

It's allright. Sometimes we have a hectic life haha. I will keep u updated as soon as I have some more info avout the camera

**Comment by OfficialDevvCat** [comment]
@guino I just found a HUGE thing. I connected my camera and looked for ports. Port `80` was open. And in the dumped firmware file it said something about port `80` being used as a RTSP proxy. I opened VLC and connected to `rtsp://192.168.1.146:80/main` and it worked! WOW!

**Comment by guino** [comment]
@OfficialDevvCat that's with the hacked firmware or just the default one ?

**Comment by EpicLPer** [comment]
> @guino I just found a HUGE thing. I connected my camera and looked for ports. Port `80` was open. And in the dumped firmware file it said something about port `80` being used as a RTSP proxy. I opened VLC and connected to `rtsp://192.168.1.146:80/main` and it worked! WOW!

Tried this on mine without the SD Card and it sadly doesn't work.

**Comment by OfficialDevvCat** [comment]
> > @guino I just found a HUGE thing. I connected my camera and looked for ports. Port `80` was open. And in the dumped firmware file it said something about port `80` being used as a RTSP proxy. I opened VLC and connected to `rtsp://192.168.1.146:80/main` and it worked! WOW!
> 
> Tried this on mine without the SD Card and it sadly doesn't work.

Are you 100% sure? What firmware version is it on?

**Comment by OfficialDevvCat** [comment]
Okay @guino @EpicLPer I just found the password of the camera's it is `dgiot010` as stated on https://github.com/mstanislav/phd-dissertation/blob/main/Camera%20Credentials.md this page! Success! Now I need to figure out if I can actually get telnet to open port `23`. Cause it does not want to do that with the `product.cof` file...

**Comment by OfficialDevvCat** [comment]
@guino How did you patched the `ppsapp` in `dgiot`?

**Comment by guino** [comment]
@OfficialDevvCat for 7.6.32 firmware, on dgiot I only changed the code to make it force call the RTSPSTART function in their code:
```
26893,26894c26893,26894
< 000690c0: eadd 00eb 0430 d4e5 0000 53e3 0400 a011  .....0....S.....
< 000690d0: b0ff ff1b 1800 8de2 97bc 00eb 0400 a0e1  ................
---
> 000690c0: eadd 00eb 0430 d4e5 0000 53e3 0400 a0e1  .....0....S.....
> 000690d0: b0ff ffeb 1800 8de2 97bc 00eb 0400 a0e1  ................
```
@OfficialDevvCat awesome news on telnet! I'll try with mine later.

**Comment by OfficialDevvCat** [comment]
@guino the `dgiot` file is located in `/usr/bin/dgiot`. How are we able to replace that file because `/usr` is read only cause the squashfs is read only!

**Comment by guino** [comment]
@OfficialDevvCat the dgiot file in the flash can only be modified by hardware programmer or by an upgrade file - I provided a modified flash dump with it changed, but If Telnet access is enabled I would Telnet into it, then kill the dgiot running (from flash) and run a modified version from the SD card (like we do other devices).

**Comment by OfficialDevvCat** [comment]
@guino We tried killing `dgiot` but that made it restart cause the `daemon` does that. Killing `daemon` and then `dgiot` does not work eighter.

**Comment by guino** [comment]
@OfficialDevvCat if you haven't tried yet, I'd try doing it in a script to (hopefully) be fast enough to prevent a restart of it by the deamon.

**Comment by OfficialDevvCat** [comment]
@guino I tried this in the `/mnt` folder (SD card)
```
echo 'kill -9 123 && ./dgiot' > run.sh
```
and still it rebooted...

**Comment by guino** [comment]
@OfficialDevvCat I just now tried product.conf with the indoor camera (stock firmware 7.6.32) and got telnet to enable, so that's confirmation that we can easily root the indoor camera. Now I will see if I can get dgiot to run.

**Comment by aleksandersmolowik** [comment]
I have camera with original firmware v7.6.32 and `rtsp://IP_OF_CAMERA:80/main` works!

**Comment by OfficialDevvCat** [comment]
@guino try the port 80 rtsp first. Then the second issue is that there is no sound on the rtsp stream... it sucks so much....

**Comment by guino** [comment]
@OfficialDevvCat RTSP does work by default on port 80 and 8554 for me (without any patching or even SD card). The only issue is no audio (as mentioned), looking at their code it simply doesn't seem to be doing any audio streaming on the RTSP feeds (rtsp://IP:8554/main and /sub) -- I'm not sure there's much we can do about that but I'll try to inspect the ONVIF information to see if I find anything.

**Comment by EpicLPer** [comment]
Seems the newest update removed that then, damn...

**Comment by OfficialDevvCat** [comment]
> Seems the newest update removed that then, damn...

What version are you on if I may ask?

**Comment by OfficialDevvCat** [comment]
@guino I could do some weird hackery stuff by using WebRTC. But so far the quality is SD and did not yet find a way to change it to HD. Maybe we could create a small script on the sd that transmits audio to a different stream. But I have no C knowlage at all and also do not even know where to start. This camera seems to be a big pain in the ass....

**Comment by guino** [comment]
@OfficialDevvCat my indoor camera feed on the WebRTC can be toggled between SD/HD on the web interface (overlay buttons on the image), if there is a way to provide RTSP using the webrtc I think it's definitely worth having -- I just don't know how much time it would take to get that solution working well.

**Comment by OfficialDevvCat** [comment]
@guino i cloned a official tuya repository that creates a webserver and websocket to access the camera. It does work but still not able to switch to hd. Still trying that out. Im not sure if it creates a local rtsp stream or it gets one from the tuya cloud. It would be very usefull if it creates a local rtsp stream. I will see what I can do! :)

**Comment by guino** [comment]
@OfficialDevvCat I'm writing up some information on this device for future reference and as a guide for anyone that has this (or similar) device. I intend to add two options for using motion detect notifications (mqtt) with/without changes to the firmware. Without changing the firmware the only option for this device is to make a script (to be run on a separate machine) that logs in by telnet and executes an 'init' script. I was unable to find any options to run a script from the SD card without modifying the flash (which will be the 2nd option). I was also unable to find a dedicated jpeg buffer for snap/mjpeg, as the code on this device seems to only request the jpg encoding on demand (by event).

**Comment by guino** [comment]
This is the initial information for this device: https://github.com/guino/LSC1080P -- my device is currently bricked from trying to flash changes from telnet. I'll wait for my heat gun to arrive so I can repair and try some more stuff from telnet.

**Comment by OfficialDevvCat** [comment]
> This is the initial information for this device: https://github.com/guino/LSC1080P -- my device is currently bricked from trying to flash changes from telnet. I'll wait for my heat gun to arrive so I can repair and try some more stuff from telnet.

Oh no. Sorry its been a while. I didn't see the notitication 6 days ago. Well I hope you're able to repair the device (most likely will) but you also should be adding to the cons in the repo that sound is not being transmitted trough RTSP. I'm willing to give C a try to create a audio stream but I will most likely fail in that as C is super hard to learn. Maybe there is some other way to do it without using the cloud.

I also have to update all my lights soon in my house to work without tuya's cloud service. But that is a whole different topic haha.

Enjoy the weekend!

**Comment by guino** [comment]
@OfficialDevvCat I thought the audio was working on the RTSP feed -- I honestly didn't double check before my device died (no success in fixing it). Without a working device it will be hard to try and patch the application to see if we can get the audio working.

**Comment by OfficialDevvCat** [comment]
> @OfficialDevvCat I thought the audio was working on the RTSP feed -- I honestly didn't double check before my device died (no success in fixing it). Without a working device it will be hard to try and patch the application to see if we can get the audio working.

Audio is only working on the webrtc stream. Not the rtsp feed. Again, we could do this trough discord call if u want to? That would make it a lot more easy!

**Comment by guino** [comment]
We could try discord/telegram -- I am traveling until the 30th then we can look into it, really wishing my device didn't die.

**Comment by guino** [comment]
@OfficialDevvCat good news: my friend visiting from Netherlands happened to have/bring one of the indoor cameras with him so I should be able to mess with the main application to try and make the RTSP audio work again -- I don't plan to mess with the flash though.

**Comment by OfficialDevvCat** [comment]
@guino cool! Awesome. Let me know if you found anything on the audio part or something else. Getting audio work would be AMAZING! Let me know if i can do anything

**Comment by guino** [comment]
I just got back from travel last few days so I should hopefully find some  time to mess with it in the next few days.

**Comment by guino** [comment]
I haven't forgotten about trying to get audio working on the 7.6.32 firmware -- I've just been working on the outdoor and rotating camera lately (https://github.com/guino/Merkury1080P/issues/42#issuecomment-1338738051) just wanted to provide some feedback on this one.

**Comment by guino** [comment]
Sad news for anyone waiting on this one: There's no RTSP code with audio in this firmware -- even in the 'unused code' of the firmware. If you need RTSP with audio this camera is a bad option. It is possible that a future firmware update may include some test RTSP code with audio and we could then enable it for general use.

**Comment by OfficialDevvCat** [comment]
@guino Glad ti hesr from you again! The best wishes in advance and a happy new year!

Sad to hear that there is no audio in the RTSP code. I think I will be ditching these cameras and buy other ones. It's nice to have a cheap option as a security camera. But if you want audio, this is not the way to go. Being dependent on the cloud is not what I want. Maybe others, but is not reliable anyway.

I appreciate all the help you did on this camera! It was a very fun journey, and it was nice meeting you, talking with you about this, and learning new stuff.

**Comment by guino** [comment]
@OfficialDevvCat I was also able to convert WebRTC feed of the new cameras into RTSP using https://github.com/koush/scrypted -- you do need a machine with docker running to do the conversion/proxy and you need to create a free dev account on tuya server to allow access which is bit of an involved process. While it does work on all newer cameras and it does provide RTSP with audio it only works in low resolution like you said. So that is basically the same thing as the webrtc sample you tried.

It does seem worth trying to find out how to switch the resolution of the WebRTC feed since all newer devices seem to support it.

Looking at their webrtc portal implementation (protect-us.ismartlife.me) it seems that they send this message to switch to high resolution:

```
{
  "protocol": 312,
  "pv": "2.2",
  "t": 1674771813933,
  "data": {
    "header": {
      "from": "AZ1674767619------------------xZAkIpZG1YQDc",
      "to": "ebd36----------gn",
      "sub_dev_id": "",
      "sessionid": "c741033-----------------aabaa4fd9dd",
      "moto_id": "signaling14705",
      "seq": 0,
      "rtx": 0,
      "type": "resolution",
      "tid": "7627e7------------2db6911c"
    },
    "msg": {
      "mode": "webrtc",
      "cmdValue": 0
    }
  }
}
```

The "cmdValue" is the key thing here were 0=HD and 1=SD (default).

Since their sample webrtc client already generates similar messages I would not expect to be too hard to add the message to switch the resolution -- here's a sample "disconnect" message from their portal:

```
{
  "protocol": 302,
  "pv": "2.2",
  "t": 1674772074815,
  "data": {
    "header": {
      "from": "AZ1674767619------------------xZAkIpZG1YQDc",
      "to": "ebd36----------gn",
      "sub_dev_id": "",
      "sessionid": "c741033-----------------aabaa4fd9dd",
      "moto_id": "signaling14705",
      "seq": 0,
      "rtx": 0,
      "type": "disconnect"
    },
    "msg": {}
  }
}
```

Since the webrtc sample code they provided also sends a 'disconnect' type message the first thing I would try is to create a function that sends message type "resolution" with the payload:
```
"mode": "webrtc",
"cmdValue": 0
```

I personally don't enjoy javascript/web development so I'm not sure if I'm going to take the time to try it out but if you already have their sample working it could be worth a try and/or do some debugging to compare the messages the sample sends with what the messages their portal sends.

It is possible (even likely) they use a different API for their webrtc portal than what they published, so it may be required to replicate the whole new api which would be too much work.

**Comment by mehmetahsen** [comment]
Hello! I had my parallel journey with this camera, dumping the flash, disassembling the code, and indeed `dgiot` mounts the sdcard at `/mnt` and uses `product.cof` if it's there. `daemon` is just a flasher waiting for ota flag in /tmp. Also there can be a `[STATION]` section in `product.cof` which can have wifi creds in `stationssid` and `stationpwd` keys.

I've turned on the camera and let it format the sdcard, unplugged, and put the aforementioned `product.cof`. Telnet with `root:dgiot010` was a success:

```
$ telnet 192.168.x.x
(none) login: root
Password: 
BusyBox v1.26.2 (2021-10-25 16:05:04 CST) built-in shell (ash)
Enter 'help' for a list of built-in commands.
# cat /proc/cpuinfo 
processor       : 0
model name      : ARMv6-compatible processor rev 7 (v6l)
BogoMIPS        : 597.60
Features        : half thumb fastmult vfp edsp java tls 
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xb76
CPU revision    : 7
Hardware        : FH8626V100
Revision        : 0000
Serial          : 0000000000000000
```
[product.cof.zip](https://github.com/guino/Merkury1080P/files/10542281/product.cof.zip)

As `daemon` seems to be a flasher waiting for a flag and firmware file, maybe we could use the same mechanism to flash?

**Comment by guino** [comment]
@mehmetahsen we played a little with the station settings initially too, but these only work if the wifi is in 'station' mode which is not default when we were trying to use some of the other settings/files.

I'm sure there's a way to flash the device using their tools (when I tried it without their tools I corrupted my flash). On the outdoor/rotating camera all it takes is copying a .bin file to the root of the SD card and booting it up (the .bin was provided by them but I'm sure it can be customized). It's just a matter of spending time to define the method/file structure, etc. 

Due to the lack of audio in the RTSP code of this device, it is not really high in my priority list. The WebRTC<->RTSP conversion is what would be more interesting for this device since it requires no hacking or SD card to work (and has audio). I do agree that flashing something like OpenIPC would be perfect for it (if supported). 

Thanks for sharing the full product.cof file format -- I assume some people may want to use some of the settings in there.

**Comment by OfficialBrickJS** [comment]
Hey there @guino and @mehmetahsen,

I changed my github name so don't worry I'm some other guy haha. So here it the thing. I can take a look if I can use WebRTC to create a RTSP stream (Not RTSPS cause that is not supported by many platforms, in this case AgentDVR). But I'm not sure if it is worth it. The main reason is that it does not run locally. What is Tuya goes down? I mean, if I have such things like that in mind I would've created my own ESP8266 HD RTSP Camera or something like that. But I think doing it trough their cloud would be an options. But only if it supports HD and is stable! I will take a look if I can do something here as I am a fullstack developer.

Also @mehmetahsen I'm glad that you've got everything working! I just wish the `dgiot` was open source or atleast included with an audio stream. Also it would be amazing if we could flash `OpenIPC`. I did take a look at that a while ago and they are in `NEQ` (https://openipc.org/stages-of-firmware-development). They do have other chips that are ALMOST the same as the one we're using (https://openipc.org/supported-hardware/full-list) it is likely we could do it ourselfs POSSIBLY. I will take a look at this soon!

~ Trust Old BricksJS ~

**Comment by guino** [comment]
@OfficialBrickJS I don't expect tuya to go down anytime soon -- they have so many devices and continue to expand at a fast pace so it is likely they'll be around past the useful life of most of their devices running today. That said I agree that an offline solution would be best -- the only reason I think the online solution is good is that it can support any of their webrtc cameras without any changes, and it seems like going forward they're making all cameras with support for webrtc (so this could become a go-to TUYA RTSP solution for any newer camera). I have seen some solutions but none with the HD stream, so if you can get that working it would definitely set your project apart from the others.

**Comment by OfficialBrickJS** [comment]
@guino I do agree on all of those statements! I will try what I can this weekend if I have time. Else it will be next week! Can I ask you how you made the RTSP stream working exactly? I'm not sure if you gave me the full explaination a few post above or just a small summary. And I assume that we have to create a docker container for ever single camera that you're running? I will try what I can do with that github source you gave me. If that is the soluton to get the WebRTC to RTSP then I can make a github repo with all the information on how to set it up and how it works etc. etc. If I need any support I will be here once again haha.

Enjoy the weekend ya'll! :)

**Comment by guino** [comment]
@OfficialBricksJS to get RTSP I just used the steps provided in the scrypted tool -- basically install it (I used the docker method on raspberry pi) then created a dev account on tuya and configured it on the scrypted web interface/settings. It wasn't 'very-easy' but it wasn't that hard and once you do it for one camera it is very easy to add more cameras (within the same docker/instance of scrypted). Basically the tool has plugins one of which is tuya and it can 'stream' the WebRTC data from the tuya cameras, then it has other plugins incluging a 'Rebroadcast plugin' (which is enabled by default), so I set it to prebuffer the 'CloutRTSP', set the container to RTSP and just used the provided rtsp URL in the settings in VLC (something like: rtsp://localhost:41705/2cc7973fb1381823 ). It also provides a separate 'muted' url if desired. It is one setup for multiple cameras, so it would be ok for someone with multiple cameras or if you already have a pi running for other stuff (like me). It didn't seem to use that much CPU, but it does use ffmpeg in the background (likely without re-encoding of video). 

Most of the code is typescript so I'm assuming that modifying https://github.com/koush/scrypted/blob/main/plugins/tuya/src/camera.ts we could potentially try to add a request to the server to switch from SD to HD (just like their portal site does).

**Comment by OfficialBricksJS** [comment]
@guino Okay, so here is an update on what I already achieved. I installed Scrypted and installed the Tuya plugin and everything is working. I had some issues getting the RTSP stream working but that has been resolved. I am using AgentDVR to see all of my cameras and it has an options to add a microphone through an RTSP stream and link it to a camera. When i did that I had a local camera feed with a cloud sound feed. SO MY AUDIO FINALLY WORKS!! So at the moment the issue is fixed on my end! I will keep it this way for my environment cause getting a HD cloud feed requires more bandwith over my internet connection which is not needed when I have the local HD feed.

I will be checking out a way to alter the tuya plugin to get HD audio for the people who do want to use it. I'm not sure on how I will be achieving a fast development environment but that is something I will figure out for sure! After I got that working I will try to make a pull request to their repo so they can update it to HD audio! I hope they will implement it themselfs instead of me forking the whole stuff and making it an unofficial unmaintained project. We shall see about everything and how it will end up.

So far, these Tuya devices have been a pain in the ass and I do recommend them in a certian way. But Tuya really needs to change some things to make sales much bigger. I know that the cloud storage function is a good sell point. But I think it would be better to make it only target users who are not that tech savy and want a simple plug and play solution! But they also need to make it more open to the developers and tinkerers. Maybe up the price by even 5 euros (20 -> 25) per camera is still a pretty good deal in my oppinion!

Anyway, I will let you guys know if I get another update!

**Comment by keerttttt** [comment]
I have the same camera. so Telnet access is pretty simple. Pretty weird though that if i edit the /etc/conf/product.cof and add the "telnet=1" on the end, telnet is still not enabled. Only via the sd card it accepts it... Weird.. (I rather not use a SD card for it).

Also, so the dgiot does recording to file and also the RTSP server it seems? Also it does some watchdog stuff etc.
Can't we start another RTSP server on our own with or own command so we can control the bitrate and also add audio support?

**Comment by guino** [comment]
@keerttttt the reason it only works with the SD card is because the startup scripts (written by the manufacturer) basically copies that file when the device boots (from the SD card).

Technically it is possible to NOT run dgiot (at all) and instead run a custom written application to read the video+audio data and create the RTSP server for it. With the OS sources and right SDK this would not be so hard but without those it would take a long time to write anything since it has to match the right hardware and specific sensors. Trying to write something to run along with dgiot (in parallel) would have different challenges such as memory/cpu limitations as well as trying to synchronizing reading the video/audio buffers between the two applications.

**Comment by BreadJS** [comment]
Okay quick update on how my experience went for the last couple of months. It was good at first! Audio with rtsp stream in AgentDVR with Scrypted. And video trough local video rtsp stream. But later on there became a HUGE delay on the audio. I have no clue why this happened. Still trying to figure it out. Having a huge delay on the audio is not something what I want while I'm recording my camera's. I'm going for a new approach which will be creating an application that runs on the camera to stream the audio. I will be using ChatGPT as I have 0 experience in C/CPP or doing any of this on such hardware. I will let you guys know if I succeed or fail.

**Comment by BreadJS** [comment]
@guino I have a question. I was able to solder back the flash chip (very sketchy) on my LSC Rotateable camera and I flashed the wrong firmware on it. I do get lots of data (see file). I also cannot f
[BootUARTData.txt](https://github.com/guino/Merkury1080P/files/11399357/BootUARTData.txt)
ind the firmware that I backed up from that chip... I also cannot remove the chip cause there is a 90% chance I will break it again and maybe not even able to get it working again.

I would mean the world to me if you could help me out with this one

**Comment by guino** [comment]
@BreadJS your boot log indicates the flash firmware is either corrupt or the boot settings are incorrect, currently you have this: 
`console=ttySAK0,115200n8 root=/dev/mtdblock5 rootfstype=squashfs init=/sbin/init mtdparts=spi0.0:200K@0x0(UBOOT),4K@0x32000(ENV),4K@0x33000(ENVBK),48K@0x34000(DTB),1536K@0x40000(KERNEL),1024K@0x1C0000(ROOTFS),256K@0x2C0000(CONFIG),5120K@0x300000(APP) mem=64M memsize=64M` 

This is what I have:
`console=ttySAK0,115200n8 root=/dev/mtdblock5 rootfstype=squashfs init=/sbin/init mtdparts=spi0.0:200K@0x0(UBOOT),4K@0x32000(ENV),4K@0x33000(ENVBK),48K@0x34000(DTB),1536K@0x40000(KERNEL),1024K@0x1C0000(ROOTFS),256K@0x2C0000(CONFIG),5120K@0x300000(APP) mem=64M memsize=64M`

Since it looks exactly the same, the only conclusion is that your rootfs is corrupted.

There's a publicly available rootfs for this device as posted [here](https://fireware.tuyaeu.com:1443/smart/firmware/upgrade/ay1541668973821t35pE/165966791961ed11009a7.bin) and usually just saving the .bin file to the SD card and booting it up causes the firmware to automatically update from the file -- problem is: this requires the device to be operational (yours isn't). I haven't taken the device apart myself to see if you can get into o the bootloader from the serial console, but if you're able to pause the boot and go into bootloader prompt you may be able to flash it from there (manually), otherwise you'd have to re-flash the chip with a programmer (which you said may damage the device permanently). 

On devices that have an interactive bootloader you can usually just press a key during the boot (before kernel loads) and it will either display the boot loader prompt or ask for a password -- if you can get either of these send me an email and I can give you some pointers.

On reflashing using a programmer, if it's any easier to just disconnect pin 6 (CLK) you may be able to use the hardware programmer without removing the chip completely from the board (hopefully without causing permanent damage).

Your boot loader and kernel seem to be ok, it's just the rootfs that seems to be corrupted, so if you can just get that section flashed you should be ok at least to completely boot the device and use if offline. For using it with the Tuya cloud your data partition with cloud certificates would have to be ok -- getting it from another device would basically mean the two devices would never be allowed to be online at the same time, so it's not something that can be shared.

**Comment by guino** [comment]
@BreadJS one additional note: The update .bin file is actually a .tar file with 1024 bytes added to the end with a few pieces of information -- so I recommend using binwalk to extract the rootfs from it as you can't just copy the file into the flash as-is.

**Comment by BreadJS** [comment]
@guino Thank you very much for replying. Is there maybe a way to flash with the sd card? On boot (beginning) you can see this 
```
mmc/sd share pin!
mmc_start_init: init OK!
cdh:sd card, mmc->capacity_user:0x1e0000000 blocks!
cdh:mmc->capacity:0x1e0000000 !
cdh:test_part_dos read ok!
cdh:test_part_dos DOS_PART_MAGIC_OFFSET ok!
cdh:test_part_dos DOS_MBR ok!
reading ht_recover_fw.conf
** Unable to read file ht_recover_fw.conf **
ht_recover_fw.conf not found
KERNEL: size:0x00180000, offset:0x00040000
```
I tried copying that file and replace the name with `ht_recovery_fw.conf` and then I got this ouput:
```
mmc/sd share pin!
mmc_start_init: init OK!
cdh:sd card, mmc->capacity_user:0x1e0000000 blocks!
cdh:mmc->capacity:0x1e0000000 !
cdh:test_part_dos read ok!
cdh:test_part_dos DOS_PART_MAGIC_OFFSET ok!
cdh:test_part_dos DOS_MBR ok!
reading ht_recover_fw.conf
4363264 bytes read

magic#2e727375
type#1752396147
uboot size#52
kernel size#0
rootfs size#0
usrfs size#0
flash size#0
version#0.0.0.0
Image bad MAGIC
fw header not valid
KERNEL: size:0x00180000, offset:0x00040000

SF: 1572864 bytes @ 0x40000 Read: OK
```
And then it reboots: LOG HERE: Maybe it is different??
[boot2.txt](https://github.com/guino/Merkury1080P/files/11402662/boot2.txt)

**Comment by BreadJS** [comment]
I also want to let you know that this camera has a different CPU compared to the cpu tried to hack in this official topic. i also tried CTRL+C and pressing any key while booting, no results.... I will send you a picture of how terrible the soldering job is.

**Comment by BreadJS** [comment]
![20230505_032624.jpg](https://user-images.githubusercontent.com/83626012/236361016-640b22fe-79e3-455e-aeee-2bc942f28fb7.jpg)

**Comment by guino** [comment]
@BreadJS I have not reviewed the bootloader code in that camera. Likely it is looking parameters inside the ht_recovery_fw.conf file about a possible firmware update file -- the question is figuring out the format it is expecting. It could be as simple as just being the filename in the SD card that has the firmware update (the one I posted in the link) or it could require specific data like update=filename.bin, etc. It could even be as simple as just leaving the original .bin file in the SD card with an empty ht_recovery_fw.conf file just to 'signal' that you want it to do a firmware update. It takes a fair amount of time to review the bootloader code in ghidra because it's basically a guessing game for the load address.

On a quick string search it may be looking for a parameter: ht_fw_version=0x%02u%02u%02u%02u to compare versions so it decides to update or not. It may also be looking for the file named 'flash.bin' -- these are all guesses. There are probably many checks in place so I doubt it will do anything unless all the parameters are correct (and you should be able to check on the serial output).

**Comment by BreadJS** [comment]
I asked chat GPT and this is what I got. Possibly a solution
![image](https://user-images.githubusercontent.com/83626012/236368251-974c4ec9-2f4a-4e8d-ab82-9cdafefcf59f.png)

**Comment by BreadJS** [comment]
I cannot get it to work at all and it is driving me crazy... I'm tempted to buy another camera and returning this broken one in the newly bought box.

**Comment by guino** [comment]
@BreadJS there's hardly any way they'd be able to tell it was your fault that the device isn't working (unless you physically modified it) -- I have in fact had devices go bad without doing any modifications to them. If it's right/wrong is a different story, but these mass produced devices are usually priced knowing that a % will fail and have to be replaced for free, so chances are that device will just be tossed into the garbage after a return.

**Comment by BreadJS** [comment]
Yeah I think so too! So thank you at least and you've made me learn a lot of information about these cameras and infrastructures! lets hope that getting RTSP working on the next camera will be more easy now that there is more information available!

**Comment by mehmetahsen** [comment]
> @mehmetahsen we played a little with the station settings initially too, but these only work if the wifi is in 'station' mode which is not default when we were trying to use some of the other settings/files.
> 
> I'm sure there's a way to flash the device using their tools (when I tried it without their tools I corrupted my flash). On the outdoor/rotating camera all it takes is copying a .bin file to the root of the SD card and booting it up (the .bin was provided by them but I'm sure it can be customized). It's just a matter of spending time to define the method/file structure, etc.
> 
> Due to the lack of audio in the RTSP code of this device, it is not really high in my priority list. The WebRTC<->RTSP conversion is what would be more interesting for this device since it requires no hacking or SD card to work (and has audio). I do agree that flashing something like OpenIPC would be perfect for it (if supported).
> 
> Thanks for sharing the full product.cof file format -- I assume some people may want to use some of the settings in there.

Do you have that .bin file?

**Comment by guino** [comment]
@mehmetahsen I have no 'update' firmware bin file for this device, but I do have the modified flash dump I tried to flash (and corrupted my firmware).

**Comment by mehmetahsen** [comment]
> 

Is that the one you tried over telnet?
I think we could find a way to boot the device from sdcard, likely there is a way from uboot. It does look for a linux.bin, and that would allow us to iterate fast without the fear of bricking, alas uboot sectors are fine. I'm thinking of checking the android app, maybe we could get the firmware update by tricking it to think camera is running an older version.

**Comment by guino** [comment]
@mehmetahsen I think someone tried to get access to u-boot without success on this device, so you'd have to flash it just to get some sort of bootloader control. 

I have been able to get the firmware update download link from some devices by adjusting the version of the firmware in the main application of the device (ppsapp) -- since this device can be rooted (https://github.com/guino/LSC1080P) you could try to do the same in the dgiot application -- I personally have never tried it. You would most definitely have to be monitoring the application log to have any chance of seeing the link.

There's probably a way to 'simulate' the communication with the tuya servers (using secret keys, etc) and get a download link but those approaches take way too much time and don't always pan out.

**Comment by BreadJS** [comment]
@guino I swapped the 360 rotating camera and have a NEW WORKING one. all though the only port open is 6668 and no 80 or rtsp port. What should I do about this? Do I have to dump the firmware once again? Cause I REALLY DO NOT WANT TO DO THAT AGAIN LOL

**Comment by mehmetahsen** [comment]
> @guino I swapped the 360 rotating camera and have a NEW WORKING one. all though the only port open is 6668 and no 80 or rtsp port. What should I do about this? Do I have to dump the firmware once again? Cause I REALLY DO NOT WANT TO DO THAT AGAIN LOL

You can dump /dev/mtdblock*  to sdcard via telnet using `dd`. It's soft, safe and you get the images already partitioned!
I found the possible magic for the firmware bin, it is `0x55AA55AA`.  HW platform identifier is `FUHAN8626`

**Comment by guino** [comment]
@BreadJS for the rotating camera you just have to follow https://github.com/guino/LSCOutdoor1080P there is no risk as there’s no firmware changes required to root it.

**Comment by BreadJS** [comment]
@guino And this works on firmware 3.10.56? Cause so far telnet is notr working

**Comment by guino** [comment]
@BreadJS all cameras I have seen with 3.x firmware do NOT run linux so there’s nothing we can do with then). I thought you got a rotating camera as posted in that link (2.10.x firmware).

**Comment by BreadJS** [comment]
> @BreadJS all cameras I have seen with 3.x firmware do NOT run linux so there’s nothing we can do with then). I thought you got a rotating camera as posted in that link (2.10.x firmware).

That was the old camera. I bought a new one and returned the old one. I could open up the new one to see the board and see the tx data. And my normal cameras are running 7.6.32 (the one discussed here).

**Comment by BreadJS** [comment]
@guino ^^^

**Comment by guino** [comment]
@BreadJS someone just confirmed that 3.10.57  worked with option 3 (just running 2.10.36 ppsapp without modifying the flash), I would think it’s worth a try it on your 3.10.56 firmware.

**Comment by ThomasRi333** [comment]
There is no need for flasching this device, just put all the files (your 7632 files, the modified ppsFactory and busybox in the root of the SD-Card ... rtsp://ip:8554/main  works like a charm !
THANKS A LOT!

**Comment by BreadJS** [comment]
> There is no need for flashing this device (V7 firmware), just put all the files (your 7632 files, the modified ppsFactory and busybox in the root of the SD-Card ... rtsp://ip:8554/main  works like a charm !
> THANKS A LOT!

What files are you talking about?

**Comment by ThomasRi333** [comment]
[ppsFactoryTool.txt](https://github.com/guino/Merkury1080P/files/13949187/ppsFactoryTool.txt)
[7632.zip](https://github.com/guino/Merkury1080P/files/13949191/7632.zip)

**Comment by ThomasRi333** [comment]
Its without audio, but i dont need it ...

**Comment by kubonm** [comment]
Hello, I got LCS Indoor 1080P from Action yesterday, FW is 6.2802.61.
Unfortunately there is no RTSP, also copying ppsFactoryTool.txt, mmc content and busybox from 720P didn't helped.
Do you have any suggestions?
Thanks

**Comment by xarxas** [comment]
@kubonm Hey, have you found solution for FW 6.2802.61 ?
Thanks

**Comment by kubonm** [comment]
Unfortunately not..
Since I am using go2rtc, they added support for Tuya cameras recently so I am using it trough cloud
